src/Http/Client.php

```diff
@@ -80,7 +102,21 @@ public function request(string $method, string $path, $requestData = null, ?arra
         }
 
         $body = $request->getBody();
-        assert(! is_iterable($body));
+        if (is_iterable($body)) {
```

Comment:
Shouldn't be handled by a decorator (`BodyConverterDecorator` maybe) ?
Anyway, the body converter decorator is missing the `http_build_query` part, I think we should add that part on the decorator too?

Reply:
Added the `http_build_query` routine to the Decorator. 

Even if this type of decorator may be already been added to the Client decorator list it should be not enough to say that the request body is not iterable anymore (may be converted again by another decorator). That's why a double-check on the iterable body has been added to that section.

A little smell is still in there: since BodyConverter decoration is deferrable, at the moment the only secure-way to assert that the body has been converted to a valid `?string` is by checking that the last decorator has to be of this  type. Since the BodyConverter routine may change even this statement is not statically verifiable.. This is issue is also valid if other decorators return, for example, callables (that we don't want to execute here!). 
Also all this takes a bit of marshalling/unmarshalling overhead and it's not so great, especially if the last decorator is not a BodyConverter because this will execute callables to convert them to string, with possible related bad behaviours. IMO the decorator list should preserve its configuration order (so we cannot arbitrary re-sort them) so not better ideas come to my mind 😅

Reply:
I thought about a body normalizer method to be called in case request body is not null and not a string: https://gist.github.com/alekitto/d8a738f4fa2df7e2acf6e9e935f693d0
What do you think about it? Could fit in our case?

Reply:
I think I don't get the point :) Except for calling an internal Client method that can force the convertion into a `?string` calling the BodyConverter, all my doubts are still relevant. Also the BodyConvert decorator impacts not only on the body itself but also on other Request props like headers, etc.

Reply:
Tried to improve BodyConverter dealing with valid stream resources and callables that will return strings (avoiding useless envelopment). Client still calls the BodyConverter but duplicate some logic to avoid unwanted roundtrips. Feel free to discuss them! :)

Reply:
I don't think we can avoid some duplicate checks. These checks are necessary and the consequent actions are different, probably de-duplicating those checks means too much coupling.
LGTM, we can always refine this later.